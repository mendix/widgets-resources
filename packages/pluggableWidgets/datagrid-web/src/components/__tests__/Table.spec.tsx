import { shallow } from "enzyme";
import { createElement } from "react";
import { Table, TableProps } from "../Table";
import { ObjectItem } from "mendix";

describe("Table", () => {
    it("renders the structure correctly", () => {
        const component = shallow(<Table {...mockTableProps()} />);

        expect(component).toMatchSnapshot();
    });

    it("renders the structure correctly with sorting", () => {
        const component = shallow(<Table {...mockTableProps()} columnsSortable />);

        expect(component).toMatchSnapshot();
    });

    it("renders the structure correctly with resizing", () => {
        const component = shallow(<Table {...mockTableProps()} columnsResizable />);

        expect(component).toMatchSnapshot();
    });

    it("renders the structure correctly with dragging", () => {
        const component = shallow(<Table {...mockTableProps()} columnsDraggable />);

        expect(component).toMatchSnapshot();
    });

    it("renders the structure correctly with filtering", () => {
        const component = shallow(<Table {...mockTableProps()} columnsFilterable />);

        expect(component).toMatchSnapshot();
    });

    it("renders the structure correctly with hiding", () => {
        const component = shallow(<Table {...mockTableProps()} columnsHidable />);

        expect(component).toMatchSnapshot();
    });

    it("renders the structure correctly with paging", () => {
        const component = shallow(<Table {...mockTableProps()} paging />);

        expect(component).toMatchSnapshot();
    });

    it("renders the structure correctly with custom filtering", () => {
        const columns = [
            {
                header: "Test",
                hasWidgets: false,
                sortable: false,
                resizable: false,
                draggable: false,
                hidable: "no" as const,
                width: "autoFill" as const,
                size: 1,
                alignment: "left" as const
            }
        ];
        const component = shallow(<Table {...mockTableProps()} columnsFilterable columns={columns} />);

        expect(component).toMatchSnapshot();
    });

    it("renders the structure correctly with empty placeholder", () => {
        const component = shallow(
            <Table {...mockTableProps()} emptyPlaceholderRenderer={renderWrapper => renderWrapper(<div />)} />
        );

        expect(component).toMatchSnapshot();
    });

    it("renders the structure correctly with column alignments", () => {
        const columns = [
            {
                header: "Test",
                sortable: false,
                resizable: false,
                draggable: false,
                hidable: "no" as const,
                width: "autoFill" as const,
                size: 1,
                alignment: "center" as const
            },
            {
                header: "Test 2",
                sortable: false,
                resizable: false,
                draggable: false,
                hidable: "no" as const,
                width: "autoFill" as const,
                size: 1,
                alignment: "right" as const
            }
        ];

        const component = shallow(
            <Table
                {...mockTableProps()}
                columns={columns}
                cellRenderer={(renderWrapper, _, columnIndex) => renderWrapper(columns[columnIndex].header)}
            />
        );

        expect(component).toMatchSnapshot();
    });

    it("renders the structure correctly with dynamic row class", () => {
        const component = shallow(<Table {...mockTableProps()} rowClass={() => "myclass"} />);

        expect(component).toMatchSnapshot();
    });

    it("renders the structure correctly for preview when no header is provided", () => {
        const columns = [
            {
                header: "",
                sortable: false,
                resizable: false,
                draggable: false,
                hidable: "no" as const,
                width: "autoFill" as const,
                size: 1,
                alignment: "center" as const
            }
        ];
        const component = shallow(<Table {...mockTableProps()} preview columns={columns} />);

        expect(component).toMatchSnapshot();
    });

    it("renders the structure correctly with header wrapper", () => {
        const component = shallow(
            <Table
                {...mockTableProps()}
                headerWrapperRenderer={(_, header) => <div className="my-custom-header">{header}</div>}
            />
        );

        expect(component).toMatchSnapshot();
    });
});

function mockTableProps(): TableProps<ObjectItem> {
    const columns = [
        {
            header: "Test",
            sortable: false,
            resizable: false,
            draggable: false,
            hidable: "no" as const,
            width: "autoFill" as const,
            size: 1,
            alignment: "left" as const
        }
    ];
    return {
        setPage: jest.fn(),
        page: 1,
        hasMoreItems: false,
        pageSize: 10,
        columnsResizable: false,
        paging: false,
        pagingPosition: "bottom",
        columnsHidable: false,
        columnsDraggable: false,
        className: "test",
        columnsFilterable: false,
        columnsSortable: false,
        columns,
        valueForSort: () => "dummy",
        filterRenderer: () => <input type="text" value="dummy" />,
        cellRenderer: (renderWrapper, _, columnIndex) => renderWrapper(columns[columnIndex].header),
        headerWrapperRenderer: (_index, header) => header,
        data: [{ id: "123456" as any }]
    };
}
