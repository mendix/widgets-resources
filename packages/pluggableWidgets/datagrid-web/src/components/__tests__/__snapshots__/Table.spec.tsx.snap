// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Table renders the structure correctly 1`] = `
<div
  className="test"
>
  <div
    className="table"
    role="table"
  >
    <div
      className="table-header"
    />
    <InfiniteBody
      hasMoreItems={false}
      isInfinite={true}
      setPage={[MockFunction]}
      style={
        Object {
          "gridTemplateColumns": "1fr",
        }
      }
    >
      <Header
        className="align-column-left"
        column={
          Object {
            "accessor": "item",
            "alignment": "left",
            "canDrag": false,
            "canHide": false,
            "canResize": false,
            "canSort": false,
            "customFilter": null,
            "header": "Test",
            "hidden": false,
            "id": "0",
            "weight": 1,
            "width": "autoFill",
          }
        }
        dragOver=""
        draggable={false}
        filterable={false}
        hidable={false}
        isDragging={false}
        key="headers_column_0"
        resizable={false}
        resizer={
          <ColumnResizer
            onResizeEnds={[Function]}
            setColumnWidth={[Function]}
          />
        }
        setColumnOrder={[Function]}
        setDragOver={[Function]}
        setIsDragging={[Function]}
        setSortBy={[Function]}
        sortBy={Array []}
        sortable={false}
        visibleColumns={
          Array [
            Object {
              "accessor": "item",
              "alignment": "left",
              "canDrag": false,
              "canHide": false,
              "canResize": false,
              "canSort": false,
              "customFilter": null,
              "header": "Test",
              "hidden": false,
              "id": "0",
              "weight": 1,
              "width": "autoFill",
            },
          ]
        }
      />
      <div
        className="tr"
        key="row_0"
      >
        <div
          className="td td-borders"
          key="row_0_cell_0"
        >
          Test
        </div>
      </div>
    </InfiniteBody>
    <div
      className="table-footer"
    />
  </div>
</div>
`;

exports[`Table renders the structure correctly for preview when no header is provided 1`] = `
<div
  className="test"
>
  <div
    className="table"
    role="table"
  >
    <div
      className="table-header"
    />
    <InfiniteBody
      hasMoreItems={false}
      isInfinite={true}
      setPage={[MockFunction]}
      style={
        Object {
          "gridTemplateColumns": "1fr",
        }
      }
    >
      <Header
        className="align-column-center"
        column={
          Object {
            "accessor": "item",
            "alignment": "center",
            "canDrag": false,
            "canHide": false,
            "canResize": false,
            "canSort": false,
            "customFilter": null,
            "header": "",
            "hidden": false,
            "id": "0",
            "weight": 1,
            "width": "autoFill",
          }
        }
        dragOver=""
        draggable={false}
        filterable={false}
        hidable={false}
        isDragging={false}
        key="headers_column_0"
        preview={true}
        resizable={false}
        resizer={
          <ColumnResizer
            onResizeEnds={[Function]}
            setColumnWidth={[Function]}
          />
        }
        setColumnOrder={[Function]}
        setDragOver={[Function]}
        setIsDragging={[Function]}
        setSortBy={[Function]}
        sortBy={Array []}
        sortable={false}
        visibleColumns={
          Array [
            Object {
              "accessor": "item",
              "alignment": "center",
              "canDrag": false,
              "canHide": false,
              "canResize": false,
              "canSort": false,
              "customFilter": null,
              "header": "",
              "hidden": false,
              "id": "0",
              "weight": 1,
              "width": "autoFill",
            },
          ]
        }
      />
      <div
        className="tr"
        key="row_0"
      >
        <div
          className="td td-borders"
          key="row_0_cell_0"
        >
          Test
        </div>
      </div>
    </InfiniteBody>
    <div
      className="table-footer"
    />
  </div>
</div>
`;

exports[`Table renders the structure correctly with column alignments 1`] = `
<div
  className="test"
>
  <div
    className="table"
    role="table"
  >
    <div
      className="table-header"
    />
    <InfiniteBody
      hasMoreItems={false}
      isInfinite={true}
      setPage={[MockFunction]}
      style={
        Object {
          "gridTemplateColumns": "1fr 1fr",
        }
      }
    >
      <Header
        className="align-column-center"
        column={
          Object {
            "accessor": "item",
            "alignment": "center",
            "canDrag": false,
            "canHide": false,
            "canResize": false,
            "canSort": false,
            "customFilter": null,
            "header": "Test",
            "hidden": false,
            "id": "0",
            "weight": 1,
            "width": "autoFill",
          }
        }
        dragOver=""
        draggable={false}
        filterable={false}
        hidable={false}
        isDragging={false}
        key="headers_column_0"
        resizable={false}
        resizer={
          <ColumnResizer
            onResizeEnds={[Function]}
            setColumnWidth={[Function]}
          />
        }
        setColumnOrder={[Function]}
        setDragOver={[Function]}
        setIsDragging={[Function]}
        setSortBy={[Function]}
        sortBy={Array []}
        sortable={false}
        visibleColumns={
          Array [
            Object {
              "accessor": "item",
              "alignment": "center",
              "canDrag": false,
              "canHide": false,
              "canResize": false,
              "canSort": false,
              "customFilter": null,
              "header": "Test",
              "hidden": false,
              "id": "0",
              "weight": 1,
              "width": "autoFill",
            },
            Object {
              "accessor": "item",
              "alignment": "right",
              "canDrag": false,
              "canHide": false,
              "canResize": false,
              "canSort": false,
              "customFilter": null,
              "header": "Test 2",
              "hidden": false,
              "id": "1",
              "weight": 1,
              "width": "autoFill",
            },
          ]
        }
      />
      <Header
        className="align-column-right"
        column={
          Object {
            "accessor": "item",
            "alignment": "right",
            "canDrag": false,
            "canHide": false,
            "canResize": false,
            "canSort": false,
            "customFilter": null,
            "header": "Test 2",
            "hidden": false,
            "id": "1",
            "weight": 1,
            "width": "autoFill",
          }
        }
        dragOver=""
        draggable={false}
        filterable={false}
        hidable={false}
        isDragging={false}
        key="headers_column_1"
        resizable={false}
        resizer={
          <ColumnResizer
            onResizeEnds={[Function]}
            setColumnWidth={[Function]}
          />
        }
        setColumnOrder={[Function]}
        setDragOver={[Function]}
        setIsDragging={[Function]}
        setSortBy={[Function]}
        sortBy={Array []}
        sortable={false}
        visibleColumns={
          Array [
            Object {
              "accessor": "item",
              "alignment": "center",
              "canDrag": false,
              "canHide": false,
              "canResize": false,
              "canSort": false,
              "customFilter": null,
              "header": "Test",
              "hidden": false,
              "id": "0",
              "weight": 1,
              "width": "autoFill",
            },
            Object {
              "accessor": "item",
              "alignment": "right",
              "canDrag": false,
              "canHide": false,
              "canResize": false,
              "canSort": false,
              "customFilter": null,
              "header": "Test 2",
              "hidden": false,
              "id": "1",
              "weight": 1,
              "width": "autoFill",
            },
          ]
        }
      />
      <div
        className="tr"
        key="row_0"
      >
        <div
          className="td td-borders"
          key="row_0_cell_0"
        >
          Test
        </div>
        <div
          className="td td-borders"
          key="row_0_cell_1"
        >
          Test 2
        </div>
      </div>
    </InfiniteBody>
    <div
      className="table-footer"
    />
  </div>
</div>
`;

exports[`Table renders the structure correctly with custom filtering 1`] = `
<div
  className="test"
>
  <div
    className="table"
    role="table"
  >
    <div
      className="table-header"
    />
    <InfiniteBody
      hasMoreItems={false}
      isInfinite={true}
      setPage={[MockFunction]}
      style={
        Object {
          "gridTemplateColumns": "1fr",
        }
      }
    >
      <Header
        className="align-column-left"
        column={
          Object {
            "accessor": "item",
            "alignment": "left",
            "canDrag": false,
            "canHide": false,
            "canResize": false,
            "canSort": false,
            "customFilter": <input
              type="text"
              value="dummy"
            />,
            "header": "Test",
            "hidden": false,
            "id": "0",
            "weight": 1,
            "width": "autoFill",
          }
        }
        dragOver=""
        draggable={false}
        filterable={true}
        hidable={false}
        isDragging={false}
        key="headers_column_0"
        resizable={false}
        resizer={
          <ColumnResizer
            onResizeEnds={[Function]}
            setColumnWidth={[Function]}
          />
        }
        setColumnOrder={[Function]}
        setDragOver={[Function]}
        setIsDragging={[Function]}
        setSortBy={[Function]}
        sortBy={Array []}
        sortable={false}
        visibleColumns={
          Array [
            Object {
              "accessor": "item",
              "alignment": "left",
              "canDrag": false,
              "canHide": false,
              "canResize": false,
              "canSort": false,
              "customFilter": <input
                type="text"
                value="dummy"
              />,
              "header": "Test",
              "hidden": false,
              "id": "0",
              "weight": 1,
              "width": "autoFill",
            },
          ]
        }
      />
      <div
        className="tr"
        key="row_0"
      >
        <div
          className="td td-borders"
          key="row_0_cell_0"
        >
          Test
        </div>
      </div>
    </InfiniteBody>
    <div
      className="table-footer"
    />
  </div>
</div>
`;

exports[`Table renders the structure correctly with dragging 1`] = `
<div
  className="test"
>
  <div
    className="table"
    role="table"
  >
    <div
      className="table-header"
    />
    <InfiniteBody
      hasMoreItems={false}
      isInfinite={true}
      setPage={[MockFunction]}
      style={
        Object {
          "gridTemplateColumns": "1fr",
        }
      }
    >
      <Header
        className="align-column-left"
        column={
          Object {
            "accessor": "item",
            "alignment": "left",
            "canDrag": false,
            "canHide": false,
            "canResize": false,
            "canSort": false,
            "customFilter": null,
            "header": "Test",
            "hidden": false,
            "id": "0",
            "weight": 1,
            "width": "autoFill",
          }
        }
        dragOver=""
        draggable={true}
        filterable={false}
        hidable={false}
        isDragging={false}
        key="headers_column_0"
        resizable={false}
        resizer={
          <ColumnResizer
            onResizeEnds={[Function]}
            setColumnWidth={[Function]}
          />
        }
        setColumnOrder={[Function]}
        setDragOver={[Function]}
        setIsDragging={[Function]}
        setSortBy={[Function]}
        sortBy={Array []}
        sortable={false}
        visibleColumns={
          Array [
            Object {
              "accessor": "item",
              "alignment": "left",
              "canDrag": false,
              "canHide": false,
              "canResize": false,
              "canSort": false,
              "customFilter": null,
              "header": "Test",
              "hidden": false,
              "id": "0",
              "weight": 1,
              "width": "autoFill",
            },
          ]
        }
      />
      <div
        className="tr"
        key="row_0"
      >
        <div
          className="td td-borders"
          key="row_0_cell_0"
        >
          Test
        </div>
      </div>
    </InfiniteBody>
    <div
      className="table-footer"
    />
  </div>
</div>
`;

exports[`Table renders the structure correctly with dynamic row class 1`] = `
<div
  className="test"
>
  <div
    className="table"
    role="table"
  >
    <div
      className="table-header"
    />
    <InfiniteBody
      hasMoreItems={false}
      isInfinite={true}
      setPage={[MockFunction]}
      style={
        Object {
          "gridTemplateColumns": "1fr",
        }
      }
    >
      <Header
        className="align-column-left"
        column={
          Object {
            "accessor": "item",
            "alignment": "left",
            "canDrag": false,
            "canHide": false,
            "canResize": false,
            "canSort": false,
            "customFilter": null,
            "header": "Test",
            "hidden": false,
            "id": "0",
            "weight": 1,
            "width": "autoFill",
          }
        }
        dragOver=""
        draggable={false}
        filterable={false}
        hidable={false}
        isDragging={false}
        key="headers_column_0"
        resizable={false}
        resizer={
          <ColumnResizer
            onResizeEnds={[Function]}
            setColumnWidth={[Function]}
          />
        }
        setColumnOrder={[Function]}
        setDragOver={[Function]}
        setIsDragging={[Function]}
        setSortBy={[Function]}
        sortBy={Array []}
        sortable={false}
        visibleColumns={
          Array [
            Object {
              "accessor": "item",
              "alignment": "left",
              "canDrag": false,
              "canHide": false,
              "canResize": false,
              "canSort": false,
              "customFilter": null,
              "header": "Test",
              "hidden": false,
              "id": "0",
              "weight": 1,
              "width": "autoFill",
            },
          ]
        }
      />
      <div
        className="tr myclass"
        key="row_0"
      >
        <div
          className="td td-borders"
          key="row_0_cell_0"
        >
          Test
        </div>
      </div>
    </InfiniteBody>
    <div
      className="table-footer"
    />
  </div>
</div>
`;

exports[`Table renders the structure correctly with empty placeholder 1`] = `
<div
  className="test"
>
  <div
    className="table"
    role="table"
  >
    <div
      className="table-header"
    />
    <InfiniteBody
      hasMoreItems={false}
      isInfinite={true}
      setPage={[MockFunction]}
      style={
        Object {
          "gridTemplateColumns": "1fr",
        }
      }
    >
      <Header
        className="align-column-left"
        column={
          Object {
            "accessor": "item",
            "alignment": "left",
            "canDrag": false,
            "canHide": false,
            "canResize": false,
            "canSort": false,
            "customFilter": null,
            "header": "Test",
            "hidden": false,
            "id": "0",
            "weight": 1,
            "width": "autoFill",
          }
        }
        dragOver=""
        draggable={false}
        filterable={false}
        hidable={false}
        isDragging={false}
        key="headers_column_0"
        resizable={false}
        resizer={
          <ColumnResizer
            onResizeEnds={[Function]}
            setColumnWidth={[Function]}
          />
        }
        setColumnOrder={[Function]}
        setDragOver={[Function]}
        setIsDragging={[Function]}
        setSortBy={[Function]}
        sortBy={Array []}
        sortable={false}
        visibleColumns={
          Array [
            Object {
              "accessor": "item",
              "alignment": "left",
              "canDrag": false,
              "canHide": false,
              "canResize": false,
              "canSort": false,
              "customFilter": null,
              "header": "Test",
              "hidden": false,
              "id": "0",
              "weight": 1,
              "width": "autoFill",
            },
          ]
        }
      />
      <div
        className="tr"
        key="row_0"
      >
        <div
          className="td td-borders"
          key="row_0_cell_0"
        >
          Test
        </div>
      </div>
    </InfiniteBody>
    <div
      className="table-footer"
    />
  </div>
</div>
`;

exports[`Table renders the structure correctly with filtering 1`] = `
<div
  className="test"
>
  <div
    className="table"
    role="table"
  >
    <div
      className="table-header"
    />
    <InfiniteBody
      hasMoreItems={false}
      isInfinite={true}
      setPage={[MockFunction]}
      style={
        Object {
          "gridTemplateColumns": "1fr",
        }
      }
    >
      <Header
        className="align-column-left"
        column={
          Object {
            "accessor": "item",
            "alignment": "left",
            "canDrag": false,
            "canHide": false,
            "canResize": false,
            "canSort": false,
            "customFilter": <input
              type="text"
              value="dummy"
            />,
            "header": "Test",
            "hidden": false,
            "id": "0",
            "weight": 1,
            "width": "autoFill",
          }
        }
        dragOver=""
        draggable={false}
        filterable={true}
        hidable={false}
        isDragging={false}
        key="headers_column_0"
        resizable={false}
        resizer={
          <ColumnResizer
            onResizeEnds={[Function]}
            setColumnWidth={[Function]}
          />
        }
        setColumnOrder={[Function]}
        setDragOver={[Function]}
        setIsDragging={[Function]}
        setSortBy={[Function]}
        sortBy={Array []}
        sortable={false}
        visibleColumns={
          Array [
            Object {
              "accessor": "item",
              "alignment": "left",
              "canDrag": false,
              "canHide": false,
              "canResize": false,
              "canSort": false,
              "customFilter": <input
                type="text"
                value="dummy"
              />,
              "header": "Test",
              "hidden": false,
              "id": "0",
              "weight": 1,
              "width": "autoFill",
            },
          ]
        }
      />
      <div
        className="tr"
        key="row_0"
      >
        <div
          className="td td-borders"
          key="row_0_cell_0"
        >
          Test
        </div>
      </div>
    </InfiniteBody>
    <div
      className="table-footer"
    />
  </div>
</div>
`;

exports[`Table renders the structure correctly with header wrapper 1`] = `
<div
  className="test"
>
  <div
    className="table"
    role="table"
  >
    <div
      className="table-header"
    />
    <InfiniteBody
      hasMoreItems={false}
      isInfinite={true}
      setPage={[MockFunction]}
      style={
        Object {
          "gridTemplateColumns": "1fr",
        }
      }
    >
      <div
        className="my-custom-header"
      >
        <Header
          className="align-column-left"
          column={
            Object {
              "accessor": "item",
              "alignment": "left",
              "canDrag": false,
              "canHide": false,
              "canResize": false,
              "canSort": false,
              "customFilter": null,
              "header": "Test",
              "hidden": false,
              "id": "0",
              "weight": 1,
              "width": "autoFill",
            }
          }
          dragOver=""
          draggable={false}
          filterable={false}
          hidable={false}
          isDragging={false}
          key="headers_column_0"
          resizable={false}
          resizer={
            <ColumnResizer
              onResizeEnds={[Function]}
              setColumnWidth={[Function]}
            />
          }
          setColumnOrder={[Function]}
          setDragOver={[Function]}
          setIsDragging={[Function]}
          setSortBy={[Function]}
          sortBy={Array []}
          sortable={false}
          visibleColumns={
            Array [
              Object {
                "accessor": "item",
                "alignment": "left",
                "canDrag": false,
                "canHide": false,
                "canResize": false,
                "canSort": false,
                "customFilter": null,
                "header": "Test",
                "hidden": false,
                "id": "0",
                "weight": 1,
                "width": "autoFill",
              },
            ]
          }
        />
      </div>
      <div
        className="tr"
        key="row_0"
      >
        <div
          className="td td-borders"
          key="row_0_cell_0"
        >
          Test
        </div>
      </div>
    </InfiniteBody>
    <div
      className="table-footer"
    />
  </div>
</div>
`;

exports[`Table renders the structure correctly with hiding 1`] = `
<div
  className="test"
>
  <div
    className="table"
    role="table"
  >
    <div
      className="table-header"
    />
    <InfiniteBody
      hasMoreItems={false}
      isInfinite={true}
      setPage={[MockFunction]}
      style={
        Object {
          "gridTemplateColumns": "1fr fit-content(50px)",
        }
      }
    >
      <Header
        className="align-column-left"
        column={
          Object {
            "accessor": "item",
            "alignment": "left",
            "canDrag": false,
            "canHide": false,
            "canResize": false,
            "canSort": false,
            "customFilter": null,
            "header": "Test",
            "hidden": false,
            "id": "0",
            "weight": 1,
            "width": "autoFill",
          }
        }
        dragOver=""
        draggable={false}
        filterable={false}
        hidable={true}
        isDragging={false}
        key="headers_column_0"
        resizable={false}
        resizer={
          <ColumnResizer
            onResizeEnds={[Function]}
            setColumnWidth={[Function]}
          />
        }
        setColumnOrder={[Function]}
        setDragOver={[Function]}
        setIsDragging={[Function]}
        setSortBy={[Function]}
        sortBy={Array []}
        sortable={false}
        visibleColumns={
          Array [
            Object {
              "accessor": "item",
              "alignment": "left",
              "canDrag": false,
              "canHide": false,
              "canResize": false,
              "canSort": false,
              "customFilter": null,
              "header": "Test",
              "hidden": false,
              "id": "0",
              "weight": 1,
              "width": "autoFill",
            },
          ]
        }
      />
      <ColumnSelector
        columns={
          Array [
            Object {
              "accessor": "item",
              "alignment": "left",
              "canDrag": false,
              "canHide": false,
              "canResize": false,
              "canSort": false,
              "customFilter": null,
              "header": "Test",
              "hidden": false,
              "id": "0",
              "weight": 1,
              "width": "autoFill",
            },
          ]
        }
        hiddenColumns={Array []}
        setHiddenColumns={[Function]}
      />
      <div
        className="tr"
        key="row_0"
      >
        <div
          className="td td-borders"
          key="row_0_cell_0"
        >
          Test
        </div>
        <div
          className="td column-selector td-borders"
        />
      </div>
    </InfiniteBody>
    <div
      className="table-footer"
    />
  </div>
</div>
`;

exports[`Table renders the structure correctly with paging 1`] = `
<div
  className="test"
>
  <div
    className="table"
    role="table"
  >
    <div
      className="table-header"
    />
    <InfiniteBody
      hasMoreItems={false}
      isInfinite={false}
      setPage={[MockFunction]}
      style={
        Object {
          "gridTemplateColumns": "1fr",
        }
      }
    >
      <Header
        className="align-column-left"
        column={
          Object {
            "accessor": "item",
            "alignment": "left",
            "canDrag": false,
            "canHide": false,
            "canResize": false,
            "canSort": false,
            "customFilter": null,
            "header": "Test",
            "hidden": false,
            "id": "0",
            "weight": 1,
            "width": "autoFill",
          }
        }
        dragOver=""
        draggable={false}
        filterable={false}
        hidable={false}
        isDragging={false}
        key="headers_column_0"
        resizable={false}
        resizer={
          <ColumnResizer
            onResizeEnds={[Function]}
            setColumnWidth={[Function]}
          />
        }
        setColumnOrder={[Function]}
        setDragOver={[Function]}
        setIsDragging={[Function]}
        setSortBy={[Function]}
        sortBy={Array []}
        sortable={false}
        visibleColumns={
          Array [
            Object {
              "accessor": "item",
              "alignment": "left",
              "canDrag": false,
              "canHide": false,
              "canResize": false,
              "canSort": false,
              "customFilter": null,
              "header": "Test",
              "hidden": false,
              "id": "0",
              "weight": 1,
              "width": "autoFill",
            },
          ]
        }
      />
      <div
        className="tr"
        key="row_0"
      >
        <div
          className="td td-borders"
          key="row_0_cell_0"
        >
          Test
        </div>
      </div>
    </InfiniteBody>
    <div
      className="table-footer"
    >
      <Pagination
        canNextPage={false}
        canPreviousPage={true}
        gotoPage={[Function]}
        nextPage={[Function]}
        page={1}
        pageSize={10}
        previousPage={[Function]}
      />
    </div>
  </div>
</div>
`;

exports[`Table renders the structure correctly with resizing 1`] = `
<div
  className="test"
>
  <div
    className="table"
    role="table"
  >
    <div
      className="table-header"
    />
    <InfiniteBody
      hasMoreItems={false}
      isInfinite={true}
      setPage={[MockFunction]}
      style={
        Object {
          "gridTemplateColumns": "1fr",
        }
      }
    >
      <Header
        className="align-column-left"
        column={
          Object {
            "accessor": "item",
            "alignment": "left",
            "canDrag": false,
            "canHide": false,
            "canResize": false,
            "canSort": false,
            "customFilter": null,
            "header": "Test",
            "hidden": false,
            "id": "0",
            "weight": 1,
            "width": "autoFill",
          }
        }
        dragOver=""
        draggable={false}
        filterable={false}
        hidable={false}
        isDragging={false}
        key="headers_column_0"
        resizable={true}
        resizer={
          <ColumnResizer
            onResizeEnds={[Function]}
            setColumnWidth={[Function]}
          />
        }
        setColumnOrder={[Function]}
        setDragOver={[Function]}
        setIsDragging={[Function]}
        setSortBy={[Function]}
        sortBy={Array []}
        sortable={false}
        visibleColumns={
          Array [
            Object {
              "accessor": "item",
              "alignment": "left",
              "canDrag": false,
              "canHide": false,
              "canResize": false,
              "canSort": false,
              "customFilter": null,
              "header": "Test",
              "hidden": false,
              "id": "0",
              "weight": 1,
              "width": "autoFill",
            },
          ]
        }
      />
      <div
        className="tr"
        key="row_0"
      >
        <div
          className="td td-borders"
          key="row_0_cell_0"
        >
          Test
        </div>
      </div>
    </InfiniteBody>
    <div
      className="table-footer"
    />
  </div>
</div>
`;

exports[`Table renders the structure correctly with sorting 1`] = `
<div
  className="test"
>
  <div
    className="table"
    role="table"
  >
    <div
      className="table-header"
    />
    <InfiniteBody
      hasMoreItems={false}
      isInfinite={true}
      setPage={[MockFunction]}
      style={
        Object {
          "gridTemplateColumns": "1fr",
        }
      }
    >
      <Header
        className="align-column-left"
        column={
          Object {
            "accessor": "item",
            "alignment": "left",
            "canDrag": false,
            "canHide": false,
            "canResize": false,
            "canSort": false,
            "customFilter": null,
            "header": "Test",
            "hidden": false,
            "id": "0",
            "weight": 1,
            "width": "autoFill",
          }
        }
        dragOver=""
        draggable={false}
        filterable={false}
        hidable={false}
        isDragging={false}
        key="headers_column_0"
        resizable={false}
        resizer={
          <ColumnResizer
            onResizeEnds={[Function]}
            setColumnWidth={[Function]}
          />
        }
        setColumnOrder={[Function]}
        setDragOver={[Function]}
        setIsDragging={[Function]}
        setSortBy={[Function]}
        sortBy={Array []}
        sortable={true}
        visibleColumns={
          Array [
            Object {
              "accessor": "item",
              "alignment": "left",
              "canDrag": false,
              "canHide": false,
              "canResize": false,
              "canSort": false,
              "customFilter": null,
              "header": "Test",
              "hidden": false,
              "id": "0",
              "weight": 1,
              "width": "autoFill",
            },
          ]
        }
      />
      <div
        className="tr"
        key="row_0"
      >
        <div
          className="td td-borders"
          key="row_0_cell_0"
        >
          Test
        </div>
      </div>
    </InfiniteBody>
    <div
      className="table-footer"
    />
  </div>
</div>
`;
